src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/domain/entity/read/ReferralSummaryProjection.kt

```diff
@@ -0,0 +1,14 @@
+package uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.domain.entity.read
+
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.domain.entity.create.ReferralEntity
+import java.time.LocalDateTime
+import java.util.*
+
+data class ReferralSummaryProjection(
+  val referralId: UUID,
+  val courseName: String,
+  val audience: String,
```

Comment:
am a little confused on the use of `audience` here vs `audiences` elsewhere, particularly in terms of what the DTO looks like. Should the UI expect `audience` or `audiences`?

Reply:
I've made this distinction because we're only ever expecting a single value inside the `audience` column of our database projection, but when transforming it to the api model structure that we send to the UI, we need to collate audiences together.

For instance, for our test data in `R__test_data.sql`, the repository will produce the following for `organisationId` "BWN":

```
| referralId                           | courseName   | audience                 | status           | submittedOn | prisonNumber |
|--------------------------------------|--------------|--------------------------|------------------|-------------|--------------|
| 0c46ed09-170b-4c0f-aee8-a24eeaeeddaa | Super Course | Gang offence             | REFERRAL_STARTED | null        | 1uXTfdH      |
| 0c46ed09-170b-4c0f-aee8-a24eeaeeddaa | Super Course | General violence offence | REFERRAL_STARTED | null        | 1uXTfdH      |
| 0c46ed09-170b-4c0f-aee8-a24eeaeeddaa | Super Course | Extremism offence        | REFERRAL_STARTED | null        | 1uXTfdH      |
```

JPA then builds out three `ReferralSummaryProjection` objects:
```
ReferralSummaryProjection(referralId=0c46ed09-170b-4c0f-aee8-a24eeaeeddaa, courseName=Super Course, audience=Gang offence, status=REFERRAL_STARTED, submittedOn=null, prisonNumber=1uXTfdH)
ReferralSummaryProjection(referralId=0c46ed09-170b-4c0f-aee8-a24eeaeeddaa, courseName=Super Course, audience=General violence offence, status=REFERRAL_STARTED, submittedOn=null, prisonNumber=1uXTfdH)
ReferralSummaryProjection(referralId=0c46ed09-170b-4c0f-aee8-a24eeaeeddaa, courseName=Super Course, audience=Extremism offence, status=REFERRAL_STARTED, submittedOn=null, prisonNumber=1uXTfdH)
```

The service layer calls the transformer method `toApi()` on this returned collection, which produces the equivalent `ReferralSummary` model for the UI to read:

```
ReferralSummary(id=0c46ed09-170b-4c0f-aee8-a24eeaeeddaa, courseName=Super Course, audiences=[Gang offence, General violence offence, Extremism offence], status=referralStarted, prisonNumber=1uXTfdH, submittedOn=null)
```

The controller then makes the list of `ReferralSummary` objects pageable, which I'll omit here for brevity.

So it made sense to have the projection only return a single `audience` and then have the actual model use `audiences`.

Reply:
Thanks, that makes sense to me.